src/App.tsx

```diff
@@ -1,61 +1,131 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './components/types/Todo';
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const [errors, setErrors] = useState({
+    isTitleValid: true,
+    isUserSelected: true,
+  });
+
+  const addTodo = (todo: Todo) => {
+    setTodos(curr => [...curr, todo]);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const allowedCharactersRegex = /[^a-zA-Zа-яА-ЯёЁїЇіІєЄґҐ0-9 ]/g;
+    const newTitle = event.target.value.replace(allowedCharactersRegex, '');
+
+    setTitle(newTitle);
+    setErrors(prev => ({ ...prev, isTitleValid: true }));
+  };
+
+  const handleUserSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setErrors(prev => ({ ...prev, isUserSelected: true }));
+  };
+
+  const handleErrors = () => {
+    const hasErrors = {
+      isTitleValid: !!title.trim(),
+      isUserSelected: userId !== 0,
```

Comment:
The check `userId !== 0` is correct for ensuring a user is selected. Ensure that the initial state of `userId` is set to a valid user ID if necessary.